dialogue-core/src/main/java/com/palantir/dialogue/core/ConcurrencyLimitedChannel.java

```diff
@@ -125,4 +132,33 @@ public void onFailure(Throwable _throwable) {
             listener.onIgnore();
         }
     }
+
+    private double getUtilization() {
+        double inflight = limiter.getInflight();
+        double limit = limiter.getLimit();
+        return inflight / limit; // minLimit is 1 so we should never get NaN from this
+    }
+
+    static Instrumentation perHostInstrumentation(TaggedMetricRegistry registry, String uri) {
+        Meter limited = DialogueClientMetrics.of(registry).limited(ConcurrencyLimitedChannel.class.getSimpleName());
+        return new Instrumentation() {
+            @Override
+            public void markLimited() {
+                limited.mark();
+            }
+
+            @Override
+            public void registerUtilizationGauge(DoubleSupplier gauge) {
+                DialogueConcurrencylimiterMetrics.of(registry)
+                        .utilization()
+                        .host(uri)
```

Comment:
I don't think we consider uri to always be safe @carterkozak

Reply:
URIs are very much already present in DD.

Reply:
front-door proxy names aren't always safe, when requests span enclaves

Reply:
Have switched to a uriIndex to resolve this.